modules/primo/primo.module

```diff
@@ -87,6 +93,36 @@ function primo_settings_form() {
 }
 
 /**
+ * Get a configured client for the Primo Brief Search service ready for use.
+ *
+ * @return \Primo\BriefSearch\Client
+ *   The client.
+ */
+function primo_client() {
+  $handler = HandlerStack::create();
+
+  if (variable_get('primo_enable_logging')) {
+    $watchdog = new Watchdog();
+    $watchdog->setWatchdogType('primo');
+    $formatter = new MessageFormatter(
+      'Request: {method} {uri}. Response: {code} {res_header_content-type} {res_header_content-length}'
+    );
+
+    $logger = new Logger($watchdog, $formatter);
+    $handler->push($logger, 'logger');
+  }
+
+  return new Client(
+    new GuzzleClient([
+      'handler' => $handler,
+      'base_uri' => variable_get('primo_base_url')
+    ]),
+    variable_get('primo_institution_code'),
+    ip_address()
```

Comment:
Should this be the IP of the client machine? For logging?

Reply:
Ah, that was the client IP. :)

Reply:
:+1: